src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event details
+    /// </summary>
+    public class AzureMaintenanceEvent
```

Comment:
For now, if we do remove the events, let's make this internal to start with so we can change the surface area if needed in a non-breaking way :)